pinot-core/src/test/java/com/linkedin/pinot/query/pruner/TimeSegmentPrunerTest.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.query.pruner;
+
+import com.linkedin.pinot.common.request.BrokerRequest;
+import com.linkedin.pinot.common.utils.request.FilterQueryTree;
+import com.linkedin.pinot.common.utils.request.RequestUtils;
+import com.linkedin.pinot.core.query.pruner.TimeSegmentPruner;
+import com.linkedin.pinot.pql.parsers.Pql2Compiler;
+import junit.framework.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+/**
+ * Unit test for {@link com.linkedin.pinot.core.query.pruner.TimeSegmentPruner} class.
+ */
+public class TimeSegmentPrunerTest {
+  private static final String TIME_COLUMN_NAME = "time";
+  Pql2Compiler _compiler;
+
+  @BeforeClass
+  public void setup() {
+    _compiler = new Pql2Compiler();
+  }
+
+  @Test
+  public void test() {
+    // Equality predicate
+    String query = "select count(*) from table where time = 10";
```

Comment:
Add tests for disjoint range  < 10 or > 20 and not time column predicate

also for time < 0 or time > Long.Max